legion/legion/k8s/services.py

```diff
@@ -272,6 +273,24 @@ def deployment(self):
         self._load_deployment_data()
         return self._deployment
 
+    def _load_deployment_data_logic(self):
+        """
+        Logic (is called with retries) to load model service deployment
+
+        :return: bool
+        """
+        client = legion.k8s.utils.build_client()
+
+        extension_api = kubernetes.client.ExtensionsV1beta1Api(client)
+
+        all_deployments = extension_api.list_namespaced_deployment(self._k8s_service.metadata.namespace)
+        try:
```

Comment:
the default value - None for the function next() would be better than try/except approach. See https://docs.python.org/2/library/functions.html#next